pkg/deployment/ingester.go

```diff
@@ -0,0 +1,159 @@
+package deployment
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/apis/io/v1alpha1"
+	"github.com/jaegertracing/jaeger-operator/pkg/config/sampling"
+	"github.com/jaegertracing/jaeger-operator/pkg/service"
+	"github.com/jaegertracing/jaeger-operator/pkg/storage"
+	"github.com/jaegertracing/jaeger-operator/pkg/util"
+)
+
+// Ingester builds pods for jaegertracing/jaeger-ingester
+type Ingester struct {
+	jaeger *v1alpha1.Jaeger
+}
+
+// NewIngester builds a new Ingester struct based on the given spec
+func NewIngester(jaeger *v1alpha1.Jaeger) *Ingester {
+	if jaeger.Spec.Ingester.Size <= 0 {
+		jaeger.Spec.Ingester.Size = 1
+	}
+
+	if jaeger.Spec.Ingester.Image == "" {
+		jaeger.Spec.Ingester.Image = fmt.Sprintf("%s:%s", viper.GetString("jaeger-ingester-image"), viper.GetString("jaeger-version"))
+	}
+
+	return &Ingester{jaeger: jaeger}
+}
+
+// Get returns a ingester pod
+func (i *Ingester) Get() *appsv1.Deployment {
```

Comment:
Thinking about consistency here...

The other creators in this repo returns object not pointer. Maybe a better to standardize and include func which check if the ingester is needed

Reply:
Possibly - but the other deployments in this package (query, collector, agent) all return pointer. So this is consistent with the other deployments at the moment - so if going to be changed, it should be done in separate PR for all components.